src/main/java/com/notably/logic/commands/EditCommand.java

```diff
@@ -24,9 +26,13 @@ public EditCommand(Body body) {
      * @param notablyModel used to access the tree structure.
      * @throws CommandException
      */
-    public void execute(Model notablyModel) {
+    public void execute(Model notablyModel) throws CommandException {
         requireNonNull(notablyModel);
-        notablyModel.updateCurrentlyOpenBlockBody(body);
+        try {
+            notablyModel.updateCurrentlyOpenBlockBody(body);
+        } catch (NoSuchBlockException | CannotModifyRootException ex) {
```

Comment:
Actually I think we won't encounter any `NoSuchBlockException` here? But let's leave it just in case.

Reply:
oh yes, i've removed it.